src/routes/docs/products/ai/tutorials/music-generation/+page.markdoc

```diff
@@ -0,0 +1,172 @@
+  }
+}
+
+export default AppwriteService;
+```
+
+{% /section %}
+
+{% section #step-5 step=5 title="Create Storage bucket" %}
+In order for this function to work, create a new bucket in the Appwrite Storage. You can do this by navigating to the Appwrite Console and clicking on **Storage** in the left sidebar, then clicking on the **Create Bucket** button.
+
+{% only_dark %}
+![Create bucket on console](/images/docs/storage/dark/create-bucket.png)
+{% /only_dark %}
+
+{% only_light %}
+![Create bucket on console](/images/docs/storage/create-bucket.png)
+{% /only_light %}
+
+Use the default configuration for the bucket. Make sure to note down the bucket ID so you can add it as an environment variable later.
+{% /section %}
+
+{% section #step-6 step=6 title="Integrate with Hugging Face" %}
+in `src/main.js` implement the following function to convert text to speech using the Hugging Face API.
+
+```js
+import { fetch } from 'undici';
+import { throwIfMissing } from './utils.js';
+import AppwriteService from './appwrite.js';
+
+const HUGGINGFACE_API = 'https://api-inference.huggingface.co';
+
+export default async ({ req, res, error }) => {
+  const bucketId = process.env.APPWRITE_BUCKET_ID ?? 'generated_music';
+
+  if (!req.body.prompt || typeof req.body.prompt !== 'string') {
+    return res.json(
+      { ok: false, error: 'Missing required field `prompt`' },
+      400
+    );
+  }
+
+  const response = await fetch(
+    `${HUGGINGFACE_API}/models/facebook/musicgen-large`,
+    {
+      headers: {
+        Authorization: `Bearer ${process.env.HUGGINGFACE_ACCESS_TOKEN}`,
+      },
+      method: 'POST',
+      body: JSON.stringify({
+        inputs: req.body.prompt,
+      }),
+    }
+  );
+
+  if (!response.ok) {
+    error(await response.text());
+    return res.json({ ok: false, error: 'Failed to process text' }, 500);
+  }
+
+  const blob = await response.blob();
+  const appwrite = new AppwriteService();
+  const file = await appwrite.createFile(bucketId, blob);
+
+  return res.json({
+    ok: true,
+    fileId: file.$id,
+  });
+};
+```
+
+This Appwrite Function checks if the required environment variables are set, then processes the text using the Hugging Face API,
+stores the generated audio file in Appwrite Storage.
+{% /section %}
+
+{% section #step-7 step=7 title="Test the function" %}
+
+Test our function by sending a POST request to the function's endpoint with a JSON body containing the `prompt` parameter.
+
+Navigate to your function in the Appwrite Console and click on **Execute now**. In the modal that appears, enter the following JSON body:
+
+```json
+{
+    "prompt": "A happy tune, with a fast tempo, in the key of C major"
+}
+```
+
+Click **Execute** and you should see a response similar to the following:
+
+```json
+{
+    "ok": true,
+    "fileId": "61f7b3b3c7b7d"
+}
+```
+
+Then, use the fileId to download the generated audio file from the Appwrite Storage.
```

Comment:
Do we have support to render an audio player in the docs? Would have been cool to display the audio file too here?